narwhal/primary/src/metrics.rs

```diff
@@ -313,7 +313,10 @@ impl PrimaryChannelMetrics {
 #[derive(Clone)]
 pub struct PrimaryMetrics {
     /// count number of headers that the node proposed
-    pub headers_proposed: IntCounter,
+    pub headers_proposed: IntCounterVec,
+    // total number of parents in all proposed headers, for calculating average number of parents
+    // per header.
+    pub header_parents_total: IntCounter,
```

Comment:
You could alternatively use a `Histogram` here that would give you better break down + you can always use it to calculate avg as well

Reply:
I thought about it but setting buckets is a bit problematic. But let me update this.